controllers/hcloudremediation_controller_test.go

```diff
@@ -0,0 +1,268 @@
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "hcloud-remediation",
+				Namespace: testNs.Name,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "Machine",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       capiMachine.Name,
+						UID:        capiMachine.UID,
+					},
+				},
+			},
+			Spec: infrav1.HCloudRemediationSpec{
+				Strategy: &infrav1.RemediationStrategy{
+					Type:       "Reboot",
+					RetryLimit: 1,
+					Timeout:    &metav1.Duration{},
+				},
+			},
+		}
+
+		hcloudRemediationkey = client.ObjectKey{Namespace: testNs.Name, Name: "hcloud-remediation"}
+	})
+
+	AfterEach(func() {
+		Expect(testEnv.Cleanup(ctx, testNs, hcloudRemediation, hcloudMachine,
+			hetznerSecret, bootstrapSecret, hetznerCluster, capiMachine, capiCluster)).To(Succeed())
+	})
+
+	Context("Basic test", func() {
+		It("creates the hcloudRemediation successfully", func() {
+			Expect(testEnv.Create(ctx, hcloudRemediation)).To(Succeed())
+
+			Eventually(func() bool {
+				if err := testEnv.Get(ctx, hcloudRemediationkey, hcloudRemediation); err != nil {
+					return false
+				}
+
+				return true
+			}, timeout).Should(BeTrue())
+		})
+
+		It("checks if hcloudRemediation objects has the HCloudTokenAvailableCondition condition", func() {
+			Expect(testEnv.Create(ctx, hcloudRemediation)).To(Succeed())
+
+			Eventually(func() bool {
+				if err := testEnv.Get(ctx, hcloudRemediationkey, hcloudRemediation); err != nil {
+					return false
+				}
+
+				testEnv.GetLogger().Info("Status condition of the hcloud remediation", "condition", hcloudRemediation.Status.Conditions)
+				return isPresentAndTrue(hcloudRemediationkey, hcloudRemediation, infrav1.HCloudTokenAvailableCondition)
+			}, timeout).Should(BeTrue())
+		})
+
+		It("checks for deleting phase of hcloudRemediation object and MachineOwnerRemediatedCondition of capiMachine object", func() {
+			Eventually(func() bool {
+				if err := testEnv.Get(ctx, hcloudMachineKey, hcloudMachine); err != nil {
+					return false
+				}
+
+				testEnv.GetLogger().Info("Status of the hcloudmachine", "status", hcloudMachine.Status)
+				return hcloudMachine.Status.Ready
+			}, timeout).Should(BeTrue())
+
+			pID, err := hcloudutil.ServerIDFromProviderID(hcloudMachine.Spec.ProviderID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(hcloudClient.DeleteServer(ctx, &hcloud.Server{ID: pID})).NotTo(HaveOccurred())
+
+			Expect(testEnv.Create(ctx, hcloudRemediation)).To(Succeed())
+
+			Eventually(func() bool {
+				if err := testEnv.Get(ctx, hcloudRemediationkey, hcloudRemediation); err != nil {
+					return false
+				}
+
+				return hcloudRemediation.Status.Phase == infrav1.PhaseDeleting && isPresentAndFalseWithReason(capiMachineKey, capiMachine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason)
+			}, timeout).Should(BeTrue())
+		})
+
+		It("check if retryCount is one and LastRemediated is set", func() {
```

Comment:
It checks that, under normal conditions, a reboot is carried out and retryCound and lastRemediated are set